src/prpy/planning/vectorfield.py

```diff
@@ -240,45 +393,41 @@ def FollowVectorField(self, robot, fn_vectorfield, fn_terminate,
 
         env = robot.GetEnv()
         active_indices = robot.GetActiveDOFIndices()
-        q_limit_min, q_limit_max = robot.GetActiveDOFLimits()
+
+        # Get the robot's joint velocity limits
         qdot_limit = robot.GetDOFVelocityLimits(active_indices)
 
+        # Create a new trajectory matching the current
+        # robot's joint configuration specification
         cspec = robot.GetActiveConfigurationSpecification('linear')
         cspec.AddDeltaTimeGroup()
         cspec.ResetGroupOffsets()
-
         path = RaveCreateTrajectory(env, '')
         path.Init(cspec)
 
         time_start = time.time()
 
         def fn_wrapper(t, q):
+            """
+            The integrator will try to solve this equation
+            at each time step.
+            Note: t is the integration time and is non-monotonic.
+            """
+            # Set the joint values, without checking the joint limits
             robot.SetActiveDOFValues(q, CheckLimitsAction.Nothing)
+
             return fn_vectorfield()
 
         def fn_status_callback(t, q):
+            """
+            This is called for each collision check.
+            """
```

Comment:
This is overly specific. More than just collision checking happens in here - e.g. constraint violation checks.